sql/src/main/java/io/crate/expression/scalar/postgres/CurrentSettingFunction.java

```
@@ -0,0 +1,144 @@
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.metadata.settings.session.SessionSetting;
+import io.crate.metadata.settings.session.SessionSettingRegistry;
+import io.crate.types.DataType;
+import org.elasticsearch.common.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Locale;
+
+public class CurrentSettingFunction extends Scalar<String, Object> {
+
+    private static final String NAME = "current_setting";
+    private static final FunctionName FQN = new FunctionName(PgCatalogSchemaInfo.NAME, NAME);
+
+    public static void register(ScalarFunctionModule scalarFunctionModule) {
+        scalarFunctionModule.register(FQN,
+                                      new BaseFunctionResolver(
+                                          FuncParams
+                                              .builder(Param.STRING)
+                                              .withVarArgs(Param.BOOLEAN)
+                                              .limitVarArgOccurrences(1)
+                                              .build()) {
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> datatypes) {
+                    return new CurrentSettingFunction(
+                        new FunctionInfo(new FunctionIdent(FQN, datatypes), datatypes.get(0))
+                    );
+                }
+            });
+    }
+
+    private final FunctionInfo info;
+
+    CurrentSettingFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function function, @Nullable TransactionContext txnCtx) {
```

Isn't the default implementation good enough?

---

true, it is - Removed.

I was trying to fail early on the edge cases (`setting = null` or `missingOk = null`) 
but I guess it's not worth it.